motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -655,7 +694,138 @@ BOOL Ros_MotionServer_StopMotion(Controller* controller)
 	return(bStopped && bRet);
 }
 
+//-----------------------------------------------------------------------
+// Starts the servos
+//-----------------------------------------------------------------------
+BOOL Ros_MotionServer_StartServos(Controller* controller)
+{
+    int ret;
+	// Servo On
+	if(Ros_Controller_IsServoOn(controller) == FALSE)
+	{
+        printf("starting servos\n");
+		MP_SERVO_POWER_SEND_DATA sServoData;
+     	MP_STD_RSP_DATA rData;
+		memset(&sServoData, 0x00, sizeof(sServoData));
+   		memset(&rData, 0x00, sizeof(rData));
+		sServoData.sServoPower = 1;  // ON
+		ret = mpSetServoPower(&sServoData, &rData);
+		if( (ret == 0) && (rData.err_no ==0) )
+		{
+			// wait for the Servo On confirmation
+			int checkCount;
+			for(checkCount=0; checkCount<MOTION_START_TIMEOUT; checkCount+=MOTION_START_CHECK_PERIOD)
+			{
+				// Update status
+				Ros_Controller_StatusUpdate(controller);
+		
+				if(Ros_Controller_IsServoOn(controller) == TRUE)
+					continue;
+			
+				mpTaskDelay(MOTION_START_CHECK_PERIOD);
+			}
+			if(Ros_Controller_IsServoOn(controller) == FALSE)
+				goto updateStatus;			
+		}
+		else
+		{
+			char errMsg[ERROR_MSG_MAX_SIZE];
+			memset(errMsg, 0x00, ERROR_MSG_MAX_SIZE);
+			Ros_Controller_ErrNo_ToString(rData.err_no, errMsg, ERROR_MSG_MAX_SIZE);
+			printf("Can't turn on servo because: %s\r\n", errMsg);
+			goto updateStatus;			
+		}
+	}
+	
+updateStatus:	
+	// Update status
+	Ros_Controller_StatusUpdate(controller);
+    return Ros_Controller_IsServoOn(controller) == TRUE;
+}
 
+//-----------------------------------------------------------------------
+// Stops the servos and makes sure that the motion is also stopped
+//-----------------------------------------------------------------------
+BOOL Ros_MotionServer_StopServos(Controller* controller)
+{
+    int ret;
+    Ros_MotionServer_StopMotion(controller);

```

Comment:
Is the Ros_MotionServer_StopServos function really necessary?  There is a lot of repeat code.  Could you make one function that takes a boolean?  
Ros_MotionServer_ServoControl(Controller* controller, BOOL bEnable)